materialize-bigquery/sqlgen.go

```diff
@@ -54,9 +80,14 @@ var bqDialect = func() sql.Dialect {
 	}
 
 	return sql.Dialect{
-		Identifierer:         sql.IdentifierFn(sql.JoinTransform(".", identifierSanitizer(sql.QuoteTransform("`", "\\`")))),
-		UnquotedIdentifierer: sql.UnquotedIdentifierFn(sql.JoinTransform(".", identifierSanitizer(func(s string) string { return s }))),
-		Literaler:            sql.LiteralFn(sql.QuoteTransform("'", "\\'")),
+		Identifierer: sql.IdentifierFn(sql.JoinTransform(".",
+			identifierSanitizer(sql.PassThroughTransform(
+				func(s string) bool {
+					return simpleIdentifierRegexp.MatchString(s) && !sql.SliceContains(strings.ToUpper(s), BQ_RESERVED_WORDS)
```

Comment:
It feels a little bit off to use `ToUpper` here but `ToLower` in the Postgres equivalent.

Reply:
Yeah it's kind of weird, and really just because the list on the BigQuery docs has the reserved words in all caps. I think I'll make the common `sql.SliceContains` function do a case-insensitive comparison, since it is used in this same way for all the sql materializations. That should be more consistent.

Reply:
That sounds reasonable. This whole identifier-quoting thing should only take place during startup and the resulting queries should be cached in normal operation, so the extra work done by using `EqualFold` should be irrelevant, right?

Reply:
It's technically used in the templates for creating and updating a fence which are executed anew for every transaction.

I neglected an important detail though that's making me backpedal on thinking case-insensitivity is a good idea: We also do `SliceContains` during validation to evaluate constraints for fields. I don't actually know off-hand if flow fields are case-sensitive or not but I wouldn't feel great about enforcing that either way.

So - I think I'll actually just add a comment here, re-iterating that the reserved words list for bigquery needs to be  in all caps.